### ISSUE TITLE
Changing default viewer for PDF in User Preferences does not work

### ISSUE THREAD

**Issue by imcovangent** [jupyterlab/jupyterlab#10875]
## Description

I want to change the default PDF viewer from the standard PDF viewer to PDFJS. The options in the Settings tab clearly indicates that this is possible:

![image](https://user-images.githubusercontent.com/15325525/130033651-b770f417-2cc2-4991-8feb-790eed7f1cdb.png)

However, when I apply this setting. My PDFs still open in the standard PDF viewer and not in PDFJS.

## Reproduce

Install jupyterlab latex extension to make the PDFJS viewer available.

Go to Settings -> Advanced Settings Editor -> Document manager

Change User Preferences to:

![image](https://user-images.githubusercontent.com/15325525/130033804-2dbfa362-7c75-439b-b0ec-6ce45f05dc9a.png)

I do get this message in the browser console:

![image](https://user-images.githubusercontent.com/15325525/130033878-96f50ff6-6b56-45c4-9ee8-cb71895b459a.png)

More surprisingly, I even get this message when I set it to PDF instead of PDFJS:

![image](https://user-images.githubusercontent.com/15325525/130034064-45934d18-605c-41c9-bddb-62fcdcd56d8f.png)

![image](https://user-images.githubusercontent.com/15325525/130034030-d9d35a12-0cab-4013-b917-47d591ed87d2.png)

## Expected behavior

I would expect PDFs to be opened with the PDFJS viewer.

## Context

- Operating System and version: Ubuntu 18.04
- Browser and version: Chrome 92.0.4515.159
- JupyterLab version: 3.1.6

**Comment by krassowski** [comment]
Thank you for reporting this issue .This is a duplicate of https://github.com/jupyterlab/jupyterlab/issues/7776 which was fixed in https://github.com/jupyterlab/jupyterlab/pull/11541; currently the fix is scheduled to be included in 4.0 but maybe we can backport it.

**Comment by fcollonval** [comment]
I don't think this is a duplicate issue. But I don't know about that viewer `PDFJS`. Could you point to the extension providing it? 

From the error displayed in the console, it looks like the viewer factory is not properly set to handle `PDF` file type.

**Comment by fcollonval** [comment]
I got an idea. Could you try setting: `{"pdf": "PDFJS"}` (with the key being lower case)?

**Comment by LiberaVeritas** [comment]
I have the exact same issue. I tried setting both lowercase and uppercase, as well as on the override json. 

When I open a pdf file, it opens the default viewer (PDF), which displays blank. If I manually right click and open with PDFJS, it works perfectly. The only issue is the default viewer setting not being applied.
